extension/src/features/password_injector.rs

```diff
@@ -24,7 +24,7 @@ pub fn PasswordInjector(props: &PasswordInjectorProps) -> Html {
                 Ok(password) if !password.is_empty() && password != existing_password => {
                     context.dispatch(ContextAction::InputPassword { password })
                 }
-                Ok(password) if password.is_empty() => {
+                Ok(password) if password.is_empty() && !existing_password.is_empty() => {
```

Comment:
was this the reason of this -> `Password modal doesn't get stuck anymore`

Reply:
firefox has a session storage with an expiration where I save the password while the session is active. When that session expires, I remove the password from the extension's memory. It was a logic mistake that once it removed the password for the first time, it got caught in a loop of removing the password forever. So no matter if you input the password, the next loop was going to erase it

Reply:
so `password` is coming from the session, where is `existing_password` coming from? user input?

Reply:
existing password is the password input from the user that stays in the "state" of the frontend. While the session is active, I check for the "in memory" password instead of the session every time. I did it like this because there are version of chrome and firefox that don't support session storage and I didn't want the user to have to input the password every time they do an action while the popup is open

Reply:
Nice! Thanks for explaning it. Could you rename a bit the variables so they are a bit more clear about what they mean? `password` and `existing_password` are quite names that makes difficult to understand the context at a glance from outsiders :'( sorry for being so picky

Reply:
yes, absolutely